pkg/agent/client/client_test.go

```diff
@@ -909,3 +911,61 @@ type testClient struct {
 	entryClient  *fakeEntryClient
 	svidClient   *fakeSVIDClient
 }
+
+func Test_client_addErrorFields(t *testing.T) {
+	log := logrus.New()
+
+	type fields struct {
+		c *Config
+	}
+	type args struct {
+		err error
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   logrus.FieldLogger
+	}{
+		{
+			name: "OK_status",
+			fields: fields{
+				c: &Config{
+					Log: log,
+				},
+			},
+			args: args{
+				err: status.Error(codes.OK, ""),
+			},
+			want: log.WithFields(
+				logrus.Fields{
+					telemetry.StatusCode:    codes.OK,
+					telemetry.StatusMessage: "",
+				}).WithError(status.Error(codes.OK, "")),
+		},
+		{
+			name: "NotFound_status",
+			fields: fields{
+				c: &Config{
+					Log: log,
+				},
+			},
+			args: args{
+				err: status.Error(codes.NotFound, "NotFound"),
+			},
+			want: log.WithFields(
+				logrus.Fields{
+					telemetry.StatusCode:    codes.NotFound,
+					telemetry.StatusMessage: "NotFound",
+				}).WithError(status.Error(codes.NotFound, "NotFound")),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &client{
+				c: tt.fields.c,
+			}
+			assert.Equalf(t, tt.want, c.addErrorFields(tt.args.err), "addErrorFields(%v)", tt.args.err)
```

Comment:
As `addErrorFields()` is a private method of `client`, I think it would be better to test that each public method in `client` produces the expected error logs when the RPCs those public methods call return errors. However, I acknowledge that would significantly increase the scope of this PR because there is no testing of log messages in this struct today. We do similar log assertions in other parts of the codebase, but it looks like it was missed in this package for some reason.

If you feel up for adding that test support, it would be much appreciated since there are currently uncovered branches, and it's a bit difficult to tell that the usages of this new method are working as expected. If not, I think it would be good to create a follow-up issue and link it as a comment somewhere in this file (maybe above each test that is covering error cases).

Reply:
I've just started working on that. Please check [29961ed](https://github.com/spiffe/spire/pull/4262/commits/29961ed199fc31639c958c19182dcdcbff42d33d) for the first tests that covers some cases. I'll be adding the remaining ones.

Reply:
Please dismiss my previous comment, I need to re-write the tests due to the changes of the ex `addErrorFields` method.

Reply:
I've just updated the new tests at [7d46273](https://github.com/spiffe/spire/pull/4262/commits/7d46273bdf1e92fda70019c64c7f82397c043dfb)